libafl/src/mutators/convert.rs

```diff
@@ -0,0 +1,45 @@
+use core::marker::PhantomData;
+
+use crate::{
+    inputs::convert::ConvertInput,
+    mutators::{MutationResult, Mutator},
+    Error,
+};
+
+#[derive(Debug)]
+pub struct ConvertMutator<I, J, S, M>
+where
+    I: Default + ConvertInput<J>,
+    J: for<'a> ConvertInput<&'a I>,
+    M: Mutator<J, S>,
+{
+    mutator: M,
+    phantom: PhantomData<(I, J, S)>,
+}
+
+impl<I, J, S, M> Mutator<I, S> for ConvertMutator<I, J, S, M>
```

Comment:
Could potentially do something like this to lift byte-level mutations to UTF-8, but we could also leave that out.

Reply:
converting inputs is a more complex thing as some of them require an external source of data. for instance a bytes input needs the parsed grammar to be converted (parsed) to a bytes input. we already have components that convert inputs and take closures that can be used to define the conversion.

@addisoncrump modified the grimoire mutators to convert from bytes input